### ISSUE TITLE
[shell][v2] How to get a `Program`  instance when using `tauri-plugin-shell`

### ISSUE THREAD

**Issue by MystiPanda** [tauri-apps/plugins-workspace#934]
Hello, I'm sorry to submit this issue, which may not be considered an issue, but recently I encountered some confusing problems when trying to migrate my project to tauri v2. It may be due to my unfamiliarity with the v2 API. , so I came to submit an issue for advice.

I try to use the `open` method provided by `tauri-plugin-shell` to open a text file with vscode. In the tauri v1, I can easily create a `Program` instance through `Program::from_str`, but in the v2 version` In the tauri-plugin-shell` plugin, I don’t know how to get the `Program` instance and pass it to `open`

In short, I would like to know how to migrate the following code snippet to v2 of the `tauri-plugin-shell` plugin:

```rust
let code = "code";
let _ = match Program::from_str(code) {
    Ok(code) => open(&app.shell_scope(), path.to_string_lossy(), Some(code)),
    Err(err) => {
        log::error!(target: "app", "Can't find VScode `{err}`");
        // default open
        open(&app.shell_scope(), path.to_string_lossy(), None)
    }
};
```

Looking forward and thank you for your reply :)

**Comment by FabianLars** [comment]
Looks like we're indeed missing a re-export there, thanks for bringing this to our attention!
I can't make any code changes until Sunday/Monday but I also wonder if it makes sense to change the rust api to allow strings as a program to be more flexible, because for example `code` should not work with the current implementation.

**Comment by MystiPanda** [comment]
I also noticed that the implementation of the `from_str` method in the v2 code does not support custom programs. I wonder if there will be any modifications to this in the future.
